src/event_loop.rs

```
@@ -116,17 +113,17 @@ impl<T> EventLoopBuilder<T> {
         doc = "[`.with_android_app(app)`]: #only-available-on-android"
     )]
     #[inline]
-    pub fn build(&mut self) -> EventLoop<T> {
+    pub fn build(&mut self) -> Result<EventLoop<T>, EventLoopError> {
         if EVENT_LOOP_CREATED.swap(true, Ordering::Relaxed) {
-            panic!("Creating EventLoop multiple times is not supported.");
+            return Err(EventLoopError::AlreadyRunning);
```

Is this really recoverable? When would you ever try to build an event loop where one already exists, and not have that fail?

---

Hm, yeah, we could assert, I guess?

---

I think this should return an error, as there's no other real way to tell if the event loop has otherwise been created.

---

That was my initial thinking that you should indicate that there's a loop running, but maybe not fail hard.

The thing were we should fail hard is when `main` thread is not used, because it's assertion basically on API misuse.

---

Well, on the other side it's a user controllable action, because winit tracks its own function calls, to not allow creating event loop again on macOS, because it'll break and we don't really want to support that.

---

The issue here is that it's a logical error of the programmer, not an error coming from the outside world or interacting with it. It's just whether you've created the event loop by winit means or not.